radixdlt/babylon-wallet-android
# Adapt wallet to the changes on `HDPathComponent`

## Description
For more information about sargon's changes take a look at this [PR](https://github.com/radixdlt/sargon/pull/243).

In general a simple `HDPathValue` is no more. Values have been scoped so we can differentiate between them. 

```
0                             2^31         2^31+2^30         2^32

|--------- Unhardened----------|---------- Hardened-----------|

|-------------------Unsecurified-------------|---Securified---|
          
```

Key areas that are affected:
(1) Calculating the next derivation path of a new entity when creating it.
(2) Rederiving public keys when doing the account recovery scan
(3) Importing from legacy wallet
(4) Creating auth-signing derivation paths from transaction signing.

## How to test
1. Rederive accounts using seed phrases and ledger. Click on next too. When done so, try rederiving again with the same seed phrase. You should not see the same derived accounts appearing in the list.
2. Import from legacy wallet
